lab_2/main.py

```diff
@@ -119,9 +261,39 @@ def create_diff_report(original_text_tokens: tuple, suspicious_text_tokens: tupl
     :param accumulated_diff_stats: a dictionary with statistics for each pair of sentences
     :return: a report
     """
-    pass
+    if not isinstance(original_text_tokens, tuple) or not isinstance(suspicious_text_tokens, tuple) or \
+            not isinstance(accumulated_diff_stats, dict):
+        return ''
+    if len(original_text_tokens) < len(suspicious_text_tokens):
+        original_text_tokens += tuple([tuple([''])]) * (len(suspicious_text_tokens) - len(original_text_tokens))
+    if len(original_text_tokens) > len(suspicious_text_tokens):
+        original_text_tokens = original_text_tokens[:len(suspicious_text_tokens)]
+    report = []
+    total_plagiarism_percent = accumulated_diff_stats['text_plagiarism'] * 100
+    for number in enumerate(suspicious_text_tokens):
+        number = number[0]
+        lcs_length = accumulated_diff_stats['sentence_lcs_length'][number]
+        plagiarism_percent = accumulated_diff_stats['sentence_plagiarism'][number] * 100
+        diff_indexes = accumulated_diff_stats['difference_indexes'][number]
+        changed_original = list(original_text_tokens[number])
+        changed_suspicious = list(suspicious_text_tokens[number])
+        if diff_indexes != ((), ()):
+            for element, indexes in enumerate(diff_indexes):
+                inserts = 0
+                for insert in indexes:
+                    if element == 0:
+                        changed_suspicious.insert(insert + inserts, '|')
+                    elif element == 1:
+                        changed_original.insert(insert + inserts, '|')
+                    inserts += 1
+        report.append('- ' + ' '.join(changed_original))
+        report.append('+ ' + ' '.join(changed_suspicious))
+        report.append('lcs = {}, plagiarism = {}%'.format(lcs_length, plagiarism_percent))
+    report.append('Text average plagiarism (words): {}%'.format(total_plagiarism_percent))
+    return ' '.join(report)
 
 
+# I tried to do functions for 10 score, but they do not work correctly
```

Comment:
the optimized function seems correct, what errors do you get?

Reply:
Well, I cannot pass some tests
tokenize_big_file gives me different result
Actual tokenize_big_file function length: 36815824
Reference tokenize_big_file function length: 36970273
And find_lcs_length too
If I leave the code like that it gives me
Actual find_lcs_length_optimized function lcs is 89 
Reference find_lcs_length_optimized function lcs is 3899
But if I make a common dictionary for both texts it is still far from the reference
Actual find_lcs_length_optimized function lcs is 223
Reference find_lcs_length_optimized function lcs is 3899

Reply:
write to me